src/main/java/info/blockchain/wallet/payment/Coins.java

```diff
@@ -66,14 +102,63 @@
         return Pair.of(sweepBalance, sweepFee);
     }
 
-    public static SpendableUnspentOutputs getMinimumCoinsForPayment(UnspentOutputs coins, BigInteger paymentAmount, BigInteger feePerKb) {
+    private static ArrayList<UnspentOutput> getSortedCoins(
+        ArrayList<UnspentOutput> unspentOutputs) {
+
+        ArrayList<UnspentOutput> sortedCoins = new ArrayList<>();
+
+        //Select 1 smallest non-replayable coin
+        Collections.sort(unspentOutputs, new UnspentOutputAmountComparatorAsc());
+        for(UnspentOutput coin : unspentOutputs) {
+            if (!coin.isReplayable()) {
+                sortedCoins.add(coin);
+                break;
+            }
+        }
+
+        //Descending value. Add all replayable coins.
+        Collections.sort(unspentOutputs, new UnspentOutputAmountComparatorDesc());
```

Comment:
Does this sort need to be done again here? I think it would be cheaper to just reverse the list.

Reply:
Good catch. A reverse would be much faster thanks.